### DISCUSSION TITLE
[Feature]: Buffer Config for iOS.

### DISCUSSION THREAD

**Discussion by Michota** [TheWidlarzGroup/react-native-video#4971] [Ideas]
### What problem are you trying to solve?

It's very important for those who want to create apps without paying much for bandwidth and storage. Just like in the case of Android, users want to control buffer related things, like _back-buffer_, etc. with usage of `bufferConfig` — but this is currently not available on iOS.

### Proposed solution

Allow users to control buffer related things on iOS (like _back-buffer_, etc.) with usage of `bufferConfig`, just like on Android.

**Possible implementation**

use [`preferredForwardBufferDuration`](https://developer.apple.com/documentation/avfoundation/avplayeritem/1643630-preferredforwardbufferduration)

### Platforms

iOS

### Additional context

It seems like there is already a PR that includes that #1353

**Comment by DLowHP** [comment]
Bumping this. Would also appreciate this added.

**Comment by Michota** [comment]
Bump. I need this so badly. I would even move from Expo-AV and donate some cash if iOS buffer config will be working as intended.

**Comment by KrzysztofMoch** [comment]
@Michota it seams that AVPlayer (iOS) is very limited in this area compared to android - do you have any specific parameter that you would have ?

**Comment by Michota** [comment]
@KrzysztofMoch: [preferredForwardBufferDuration](https://developer.apple.com/documentation/avfoundation/avplayeritem/1643630-preferredforwardbufferduration?language=objc) is a way to go I think.

> This property defines the preferred **forward buffer duration in seconds**. If set to 0, the player will choose an appropriate level of buffering for most use cases. Setting this property to a low value will increase the chance that playback will stall and re-buffer, while setting it to a high value will increase demand on system resources.

**Comment by KrzysztofMoch** [comment]
@Michota there is already `preferredForwardBufferDuration` prop for you - does it not work for you?

**Comment by Michota** [comment]
Oh, i missed it.
It will work similar to how `minBufferMs` works for Android, right?

**Comment by pecabum** [comment]
It doesn't work, i have a high resolution video that should play from web and the player doesn't buffer at all on ios. I've tried all possible methods.

**Comment by emreer94** [comment]
The default is 0, it means let the system decide to buffer. I want it to buffer minimally and set it to 1 but literally nothing changed. Also, I tried this automaticallyWaitsToMinimizeStalling={false}, defaults to true, to shut down the buffering state and immediately start the streaming as indicated in properties section of the documentation but that does not work as well. How to deal with this?

**Comment by marcin-bojar** [comment]
bump, we need this as well

**Comment by TWG-AI** [comment]
v7 now has full iOS buffer configuration via `bufferConfig` in the video source:

```ts
useVideoPlayer({
  uri: 'https://example.com/stream.m3u8',
  bufferConfig: {
    preferredForwardBufferDurationMs: 10000,
    preferredPeakBitRate: 5_000_000,
    preferredMaximumResolution: { width: 1920, height: 1080 },
    livePlayback: { targetOffsetMs: 5000 },
  },
})
```

> I'm an AI — verify these parameters against the v7 docs before shipping.
